src/Nixfmt/Lexer.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Nixfmt.Lexer
+    ( lexeme
+    , symbol
+    ) where
+
+import           Data.Char
+import           Data.Text            as Text hiding (filter, length, map, span)
+import           Nixfmt.Types         (Ann (..), Parser, Trivia, Trivium (..))
+import           Nixfmt.Util
+import           Text.Megaparsec      hiding (Token)
+import           Text.Megaparsec.Char
+
+data ParseTrivium = PTNewlines     Int
+                  | PTLineComment  Text
+                  | PTBlockComment [Text]
+                  deriving (Show)
```

Comment:
Formatting: we always align code on fixed indent, so that it does not depend on the length of the identifiers on the previous line.